### DISCUSSION TITLE
Installing cs_tools on my Mac

### DISCUSSION THREAD

**Discussion by ravi-harve** [thoughtspot/cs_tools#176] [General]
I am trying to install cs_tools on my Mac, and I am getting the following error. I got a new laptop and migrated my home folder using Time Machine Restore.

~/ThoughtSpot » cat cs_tools-bootstrap-error-tmp96jrcbh3.log
INFO     | 2024-10-11T10:13:25-0400 | <stdin>:110 | Welcome to the CS Tools Bootstrapper!
         | Ideally, you will only need to run this one time, and then your environment can be fully managed by CS Tools itself.
         | If you run into any issues, please reach out to us on GitHub Discussions below.
         |
         |           GitHub: https://github.com/thoughtspot/cs_tools/issues/new/choose
         |
DEBUG    | 2024-10-11T10:13:25-0400 | <stdin>:127 |
    [PLATFORM DETAILS]
    Python Version: 3.12.6
       System Info: Darwin (detail: macOS-14.7-arm64-arm-64bit)
     Platform Tags: macosx-14.6-arm64
            Ran at: 2024-10-11 14:13:25 +0000

INFO     | 2024-10-11T10:13:25-0400 | <stdin>:439 | Missing '_updater.py', downloading from GitHub
INFO     | 2024-10-11T10:13:25-0400 | <stdin>:446 | Downloaded as '_updater.py'
INFO     | 2024-10-11T10:13:25-0400 | <stdin>:155 | Creating the CS Tools virtual environment.
DEBUG    | 2024-10-11T10:13:25-0400 | _updater.py:255 | Executing venv creation: /Users/harve/.pyenv/versions/3.12.6/bin/python3 -m venv /Users/harve/Library/Application Support/cs_tools/.cs_tools
INFO     | 2024-10-11T10:13:25-0400 | _updater.py:127 | Error: [Errno 2] No such file or directory: '/Users/harve/Library/Application Support/cs_tools/.cs_tools/bin/python3'
DEBUG    | 2024-10-11T10:13:25-0400 | <stdin>:569 | Error found: Failed with exit code: 1

PIP COMMAND BELOW
/Users/harve/.pyenv/versions/3.12.6/bin/python3 -m venv /Users/harve/Library/ApplicationSupport/cs_tools/.cs_tools
Traceback (most recent call last):
  File "<stdin>", line 564, in main
  File "<stdin>", line 156, in cli
  File "/Users/harve/ThoughtSpot/_updater.py", line 256, in make
    self.run(python, "-m", "venv", self.venv_path.as_posix())
  File "/Users/harve/ThoughtSpot/_updater.py", line 133, in run
    raise RuntimeError(f"Failed with exit code: {proc.returncode}\n\nPIP COMMAND BELOW\n{cmd}")
RuntimeError: Failed with exit code: 1

PIP COMMAND BELOW
/Users/harve/.pyenv/versions/3.12.6/bin/python3 -m venv /Users/harve/Library/ApplicationSupport/cs_tools/.cs_tools
WARNING  | 2024-10-11T10:13:25-0400 | <stdin>:570 | Unexpected error in bootstrapper, see /Users/harve/ThoughtSpot/cs_tools-bootstrap-error-tmp96jrcbh3.log for details..

**Comment by ravi-harve** [comment]
I was able to resolve the issue by removing the directory cs_tools from Application Support directory

**Reply by boonhapus** [reply]
It looks like your new environment uses `pyenv` , which will dynamically point to the venv we maintain in cs tools based on which version you've got activated. python's `venv` module just symlinks your venv executable to the then-system python's executable.

When your drive copied over to the new hardware, it's possible we had a py-version mismatch or similar. A reinstall likely would have sorted you out as well as that rebuilds the cs tools venv. Razing the directory does that same thing though! 😄
